com.unity.shadergraph/Editor/Data/Nodes/Input/Texture/ProceduralVirtualTextureNode.cs

```diff
@@ -83,6 +91,21 @@ public int layers
             }
         }
 
+        [EnumControl("Bind Method")]
+        public BindMethod bindMethod
```

Comment:
Can we use the HLSLDeclaration instead?

This probably means using a property drawer UI for the node, though, so you can filter out the HLSLDeclaration.PerInstance from the set of possibilities.

But property drawers (that show up in the node inspector) are preferred over adding more controls to the node itself, in general.

You can look at SampleTexture2DNodePropertyDrawer as an example.

And also BuildHLSLDeclarationOverrideFields() in ShaderInputPropertyDrawer.cs for how it constructs a filtered enum control for HLSLDeclarations.

Reply:
Yes, this has been updated to an `internal HLSLDeclaration shaderDeclaration`; internal as it is now exposed in the node inspector rather than on the node itself. I also stuck with "shader declaration" labeling to have it more consistent with other shader properties.